edge-functions/edge-config-statsig/middleware.ts

```diff
@@ -0,0 +1,47 @@
+import { NextRequest, NextResponse } from 'next/server'
+import Statsig from "statsig-node";
+import { EdgeConfigDataAdapter } from "statsig-node-vercel"
+import { EXPERIMENT, UID_COOKIE } from './lib/constants'
+
+// We'll use this to validate a random UUID
+const IS_UUID = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{16}$/i
+
+export async function middleware(req: NextRequest) {
+  // If the request is not for `/`, continue
+  if (req.nextUrl.pathname !== '/') return
+
+  // Get the user ID from the cookie or get a new one
+  let userId = req.cookies.get(UID_COOKIE)?.value
+  let hasUserId = !!userId
+
+  // If there's no active user ID in cookies or its value is invalid, get a new one
+  if (!userId || !IS_UUID.test(userId)) {
+    userId = crypto.randomUUID()
+    hasUserId = false
+  }
+
+  const dataAdapter = new EdgeConfigDataAdapter(process.env.EDGE_CONFIG_ITEM_KEY!);
+  await Statsig.initialize(
+    process.env.STATSIG_SERVER_API_KEY!,
+    { dataAdapter } 
+  );
+
+  const experiment = await Statsig.getExperiment({ userID: userId }, EXPERIMENT);
+
+  const bucket = experiment.get("bucket", "Experiment not set up, please read README to set up example.")
```

Comment:
Assuming `"Experiment not set up, please read README to set up example."` is the fallback value here, it doesn't make sense to actually go ahead and rewrite the request below in `url.pathname = `/${bucket}` using the fallback value.

We could either throw instead, respond with a json error message or rewrite to an error page I guess?

Reply:
redirecting to a deafult bucket that shows this error message